modules/core/src/main/java/org/apache/fluo/core/worker/finder/hash/TableRange.java

```diff
@@ -104,21 +105,12 @@ public Range getRange() {
     return new Range(tper, false, ter, true);
   }
 
-  @Override
-  public int compareTo(TableRange o) {
-    if (Objects.equals(getEndRow(), o.getEndRow())) {
-      // this will catch case of both null
-      return 0;
-    }
-
-    if (getEndRow() == null) {
-      return 1;
-    }
+  private static final Comparator<TableRange> TABLE_RANGE_COMPARATOR =
+      Comparator.comparing(TableRange::getPrevEndRow, Comparator.nullsFirst(Bytes::compareTo))
```

Comment:
It looks like the new code compares `prevEndRow` in addition to `endRow`. Is there a reason for this?

Reply:
Thought was to compare the entire object but I see now it's not needed.